flow/record/base.py

```diff
@@ -756,6 +756,30 @@ def fieldtype(clspath):
     return t
 
 
+@functools.lru_cache(maxsize=4069)
+def merge_record_descriptors(descriptors, replace=False, name=None):
```

Comment:
Maybe start adding type hints to new pieces of code? Don't need to do `base.py` in its entirety but new or modified functions?

Reply:
Yes good one, see new commit. I'm also thinking about deprecating some initialisation code of `RecordDescriptor`. But saving that for another PR.